### PR TITLE
Replace logger with print

### PR DESCRIPTION
**Pull request by FrickTobias**
See #19 

In short, this 
 - replaces `logger.info()` with `print()` and `logger.warning()` with `print("WARNING: ")`
 - keeps `logger.info()`/`logger.warning()` only for `log_file` writing

### CODE REVIEW

File: python_tsp/heuristics/local_search.py
Comment (fillipe-gsm) [inline python_tsp/heuristics/local_search.py:83]: So, instead of keeping the `logger` (which has been shown to be a major source of confusion when a root level file also set up a logger as we talked in the issue), can you replace it with a regular print but printing to a file?

It would be something like this. First, you open the file at the beginning of the execution:

```python
log_file_handler = open(log_file, "w", encoding="utf-8")
```

Then here you would call 
```suggestion
                    print(warning_msg, file=log_file_handler)
```

Then at the end of the execution before returning you close it

```python
log_file_handler.close()
```

File: python_tsp/heuristics/local_search.py
Comment (fillipe-gsm) [inline python_tsp/heuristics/local_search.py:84]: One more thing: instead of simply printing the message, can you add the `verbose` argument as you suggested initially and it set to `False` as default?

With that, the `print` would only happen if it was set to `True`:

```suggestion
                if verbose:
                    print(f"WARNING: {warning_msg}")
```

**Review by fillipe-gsm — COMMENTED** [review]
Thanks for this first version. I suggested to use `print` even when "logging" to a file, and if possible to add a `verbose` argument as you initially suggested.

The idea is these prints remain independent.

If you need I am happy to help.

**Comment by FrickTobias** [comment]
Thanks for the feedback, I changed it according to your suggestions, except one case which I thought merited discussion. 

In case of a warning, does it make sense to regulate it with the same argument as iteration prints? Maybe adding a silent mode for suppressing warnings (default `False`)? Or am I just overthinking this?

**Comment by fillipe-gsm** [comment]
> In case of a warning, does it make sense to regulate it with the same argument as iteration prints? Maybe adding a silent mode for suppressing warnings (default `False`)? Or am I just overthinking this?

That is actually a good question. But in my opinion the gain in flexibility could overcomplicate what should be simple.
I think we can consider all prints to be the same level, so we would have the rules:
- If there is a log file, print everything to the file;
- If `verbose` is `True`, print everything to the stdout. If `False`, print nothing, _even the warnings_.

What do you think?

File: python_tsp/heuristics/local_search.py
Comment (fillipe-gsm) [inline python_tsp/heuristics/local_search.py:78]: If we opt to follow the suggestions in [this comment](https://github.com/fillipe-gsm/python-tsp/pull/20#issuecomment-1194160693), then the only last suggestion would be to only print the warning if `verbose` is set to `True`:

```suggestion
                if verbose:
                    print(warning_msg)
```

Even if the warning is important, it may be weird to see something in the stdout even by setting `verbose` as `False`.

So just this modification on both files would be enough.

**Comment by FrickTobias** [comment]
I agree. Simplicity is probably better here.

**Review by fillipe-gsm — APPROVED** [review]
Thank you very much for the work and the discussion!

I will let you merge since the work was all yours. :)
As soon as it gets merged I will package a 0.3.1 version with the fix and add your contribution to the README.

Thank you once more!

**Comment by FrickTobias** [comment]
Thank you again for sharing your code and actively maintaining it!

I think you'll have to merge though since I don't have write access to the repo!

Screen shot =>
Screen Shot 2022-07-27 at 9 55 58

**Comment by fillipe-gsm** [comment]
> Thank you again for sharing your code and actively maintaining it!
> 
> I think you'll have to merge though since I don't have write access to the repo!
> 
> Screen shot => Screen Shot 2022-07-27 at 9 55 58

That's odd; I was sure an approval from an admin would be enough :confused: 
I'll look into that later. For now, I merged the PR.

Thanks again for the work!

**Pull request merged**
